prereise/gather/demanddata/eia/map_ba.py

```diff
@@ -95,3 +95,32 @@ def map_buses_to_county(bus_county_map):
         except TypeError:
             bus_no_county_match.append(index)
     return bus_county_map, bus_no_county_match
+
+
+def map_buses_to_ba(bus_gis):
+    """Find the Balancing Authority in the U.S. territory that each bus in the query grid
+    belongs to.
+
+    :param (*pandas.DataFrame*) bus_gis: data frame contains a list of
+        entries with lat and long of buses.
+    :return: (*pandas.DataFrame*) data frame of BA that buses
+        locate
+    """
+
+    b2c_key = map_buses_to_county(bus_gis)[0].iloc[:, :-1].dropna()
+
+    # read json file for BA_County Map
+    import json
+
+    with open("../../data/BA_County_map.json") as f:
+        ba_county_map = json.load(f)
+
+    ba2county = pd.DataFrame(
+        [[k, v] for k, a in ba_county_map.items() for v in a], columns=["BA", "County"]
+    )
+
+    return (
+        b2c_key.reset_index()
+        .merge(ba2county, how="left", on=["County"], suffixes=("", ""))
+        .set_index("bus_id")
+    )
```

Comment:
Instead of transforming dictionary into a data frame then trying to merge two data frames based on "County" column, we could simply reverse the dictionary (keys=County, values=BA) then `apply/map` the dictionary based on the "County" column like following:
```python
county_ba_map = {
    county: ba
    for ba, county_list in ba_county_map.items()
    for county in county_list
}
b2c_key["BA"] = b2c_key["County"].map(county_ba_map)
```
In this case, we don't need to drop column "BA" in l.110

Reply:
Again, county name mismatches will kick in at this step if there is any, i.e. the county names returned by `map_buses_to_county` may be in a different format as we have in `county_ba_map` (this is something we need to check, I'm not sure. At least, the county names are from two different sources, i.e. census API vs BA's internal documentations).